### DISCUSSION TITLE
wallet system

### DISCUSSION THREAD

**Discussion by ds-nasser** [bavix/laravel-wallet#763] [General]
there is any idea if I want to add a wallet for the system 

meaning if I have users every user has his wallet, but if i want to add a wallet for the system not related to any user like system  commission from every payment transaction

**Comment by rez1dent3** [comment]
Hello. Technically, a wallet can be assigned to any object. Need a wallet or wallets for something internal - please. The package does not check if it is a "user", it works with any object.

**Comment by justinkekeocha** [comment]
I have a question. why do we need to store balance and their decimal places. From my little research the decimal data type for MYSQL works very well with dealing with decimal place.

I do not have much experience with dealing with money and I would love if you can enlighten me.

**Reply by rez1dent3** [reply]
It's about the mantissa. I made a small example for you: https://onecompiler.com/mysql/3zmjmuwr7

You can find more information here:
https://stackoverflow.com/questions/13030368/best-data-type-to-store-money-values-in-mysql

**Reply by justinkekeocha** [reply]
Thank you max for taking out time to explain

**Reply by justinkekeocha** [reply]
I was thinking, since this conversion is very important. Shouldn't this be the default way to handle numbers in the app. Having to think whether to use HasWalletFloat, balanceFloat, depositFloat is quite a hassle and may be more susceptible to error. Maybe from the programmer forgetting to user balance or balanceFloat for example.

I saw that depositFloat uses deposit under the hood, after the conversion.

Or is there any reason why we really need two methods for deposit, where we can detect whether mantissa is present or not. I will really love to learn

**Reply by rez1dent3** [reply]
It is necessary to use one approach in the application, it is impossible to mix approaches. I don't use *Float methods in my applications. The entire float interface is prohibited via phpstan. Any reference to these methods will cause an error and the developer will have to correct it.

And now to your question. You can implement it, you can even leave only one interface, but many applications use only one pass and you don't know which one. Forcing everyone to adhere to the same approach is not correct

**Reply by justinkekeocha** [reply]
Thanks
